add flag that copies additional directories into ./vendor

otan opened the pull request:
add flag that copies additional directories into ./vendor

In the case of a package like `github.com/prometheus/client_model`,
there are protobuf files in `github.com/prometheus/client_model` which
do not get copied at all even with this tool as it is not specified in
`./vendor/modules.txt`.

Add the ability to do this by adding an `-include` flag, which
takes in `<dir_to_copy>` fields separated by commas.

(Example of this in CockroachDB: cockroachdb/cockroach#49447)

otan commented:
@pkieltyka do you mind taking a look at this? would like to put this in so we don't have to maintain our own fork of it :)

pkieltyka commented on README.md:
````diff
@@ -13,6 +13,14 @@ $ GO111MODULE=on go mod vendor
+
+```
+$ GO111MODULE=on go mod vendor
+$ modvendor -copy="**/*.c **/*.h **/*.proto" -v -additional_dirs="github.com/grpc-ecosystem/grpc-gateway:github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis/google/api,github.com/grpc-ecosystem/grpc-gateway:github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis/google/rpc,github.com/prometheus/client_model:github.com/prometheus/client_model"
````
TBH, I don't understand the syntax of "additional_dirs". I'd also prefer the term, "extra" just, `-extra=dir1` `-extra=dir2` ... or ...... single -extra flag with comma separate dirs is fine, but why are there colons in there too..?

pkieltyka replied:
or, `-include` .. ? or some other name .. should be short

otan replied:
it needs to map `package name:package directory`.

e.g.

`github.com/grpc-ecosystem/grpc-gateway` needs to include `github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis/google/api`, hence `github.com/grpc-ecosystem/grpc-gateway:github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis/google/api`.

do you have an idea of a cleaner way of representing that?

(alternatively, i guess i can just take in an array of extra directories and figure out if it belongs under any of the given packages!)

otan replied:
i've made it just comma separated by comparing prefixes against import path. works for me.

otan committed:
add flag that copies additional directories into ./vendor

In the case of a package like `github.com/prometheus/client_model`,
there are protobuf files in `github.com/prometheus/client_model` which
do not get copied at all even with this tool as it is not specified in
`./vendor/modules.txt`.

Add the ability to do this by adding an `-include` flag, which
takes in `<dir_to_copy>` fields separated by commas.

(Example of this in CockroachDB: cockroachdb/cockroach#49447)

otan commented:
thanks! do you mind cutting a new release?

pkieltyka commented:
@otan here ya go, just pushed v0.3.0
